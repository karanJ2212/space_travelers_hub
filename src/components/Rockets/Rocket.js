import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRocket } from '../../redux/rockets/rocketsSlice';

export default function Rocket() {
  const rocketData = useSelector((state) => state.rockets.rockets);

  const isLoading = useSelector((state) => state.rockets.isLoading);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRocket());
  }, [dispatch]);

  if (isLoading) {
    return <p>Loading...</p>;
  }

  return (
    <section className="rocketContainer">
      {rocketData.map((rocket) => (
        <div key={rocket.id}>
          <h2>{rocket.name}</h2>
          <img src={rocket.flickr_images[0]} alt={rocket.name} />
        </div>
      ))}
    </section>

  );
}
