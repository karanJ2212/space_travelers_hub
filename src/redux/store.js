import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsSlice';
import missionReducer from './missions/missionslice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
  },
});

export default store;
