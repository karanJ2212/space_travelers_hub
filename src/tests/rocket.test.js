import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../redux/store";
import Rocket from "../components/Rockets/Rocket";

const RocketProvider = () => (
  <Provider store={store}>
    <Rocket />
  </Provider>
);

describe("Rockets component testing", () => {
  it("checks for accurate rendering of all rockets", () => {
    const allRockets = renderer.create(<RocketProvider />).toJSON();
    expect(allRockets).toMatchSnapshot();
  });
});
