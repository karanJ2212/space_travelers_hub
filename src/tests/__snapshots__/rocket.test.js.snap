// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component testing checks for accurate rendering of all rockets 1`] = `
<div
  className="data-container"
>
  <ul
    className="map"
  />
</div>
`;
